### PR TITLE
feat: V0 Attachment compact component

### PR DESCRIPTION
**Pull request by jurokapsiar**
Adds Attachment component to react-migration-v0-v9.
Same functionality as v0 Attachment, instead of slot props uses slot components.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-migration-v0-v9/src/components/Attachment/Attachment.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      e.preventDefault();
+      return;
+    }
+    if (onClick) {
+      onClick(e);
+    }
+  }, [onClick, disabled]);
+
+  const buttonProps = useARIAButtonProps('div', {
+    onClick: handleClick,
```
Comment (ling1726) [inline packages/react-components/react-migration-v0-v9/src/components/Attachment/Attachment.tsx:34]:
```suggestion
  const buttonProps = useARIAButtonProps('div', {
    onClick: onClick,
    disabled,
```

`useARIAButtonProps` should be handlingt his for you just by passing the root props

File: packages/react-components/react-migration-v0-v9/src/components/Attachment/Attachment.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  return (
+    <div
+      ref={ref}
+      className={mergeClasses(attachmentClassName,
+        attachmentBaseClass,
+        actionable && classes.actionable,
+        className
+      )}
+      onClick={handleClick}
```
Comment (ling1726) [inline packages/react-components/react-migration-v0-v9/src/components/Attachment/Attachment.tsx:47]:
```suggestion

```

File: packages/react-components/react-migration-v0-v9/src/components/Attachment/Attachment.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      {...actionable && {
+        "data-is-focusable": true,
```
Comment (ling1726) [inline packages/react-components/react-migration-v0-v9/src/components/Attachment/Attachment.tsx:49]: why is `data-is-focusable` used? is it to keep compat with FocusZone?

File: packages/react-components/react-migration-v0-v9/src/components/Attachment/AttachmentAction.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    onKeyUp?.(e);
+  }, [onKeyUp]);
+
+  const handleKeyDown = React.useCallback(e => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      e.stopPropagation();
+    }
+    onKeyDown?.(e);
+  }, [onKeyDown]);
```
Comment (ling1726) [inline packages/react-components/react-migration-v0-v9/src/components/Attachment/AttachmentAction.tsx:31]:
```suggestion
```

From what I see the v0 implementation does not stop keyboard events bubbling to the parent attachment from the action

File: packages/react-components/react-migration-v0-v9/src/components/Attachment/Attachment.tsx (hunk 3 above)
Comment (jurokapsiar) [reply packages/react-components/react-migration-v0-v9/src/components/Attachment/Attachment.tsx:49]: yes - maybe somebody still has it inside of v0 focus zone, it is better to keep it

File: packages/react-components/react-migration-v0-v9/src/components/Attachment/AttachmentAction.tsx (hunk 4 above)
Comment (jurokapsiar) [reply packages/react-components/react-migration-v0-v9/src/components/Attachment/AttachmentAction.tsx:31]: v0 implementation does not use this keydown/keyup combination but useARIAButton does - without it pressing Enter was causing attachment onClick to be called - so I stopped it here and now it works.

**Pull request merged**